### PR TITLE
New Story Details Modal: Component Name Clean Up

### PR DESCRIPTION
**Pull request by BrittanyIRL**
## Context

When I started this, there were mandatory fields, that impacted how I named components. Now nothing is mandatory and I don't want to introduce legacy naming at the start of a feature. 

## Summary

Just eliminating "mandatory" from story details modal vocab. 

## User-facing changes

None

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

See #10644

### CODE REVIEW

**Review by samwhale — APPROVED** [review]

File: packages/story-editor/src/components/publishModal/test/content.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ const mockInputValues = {
   [INPUT_KEYS.SLUG]: '2001-space-odyssey',
 };
 
-describe('publishModal/mainContent', () => {
+describe('publishModal/content', () => {
```
Comment (BrookeGraham) [inline packages/story-editor/src/components/publishModal/test/content.js:38]: We should rename the import on line 29 `import Content from '../content';`

Comment (BrookeGraham) [reply packages/story-editor/src/components/publishModal/test/content.js:38]: And rename Component on line 71  `MainContent` to `Content`

File: packages/story-editor/src/components/publishModal/test/mainStoryInfo.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ import renderWithTheme from '../../../testUtils/renderWithTheme';
 import { ChecklistCountProvider } from '../../checklist';
 import InspectorContext from '../../inspector/context';
 import { INPUT_KEYS } from '../constants';
-import MandatoryStoryInfo from '../mainContent/mandatoryStoryInfo';
+import MainStoryInfo from '../mainContent/mainStoryInfo';
```
Comment (BrookeGraham) [inline packages/story-editor/src/components/publishModal/test/mainStoryInfo.js]:
```suggestion
import MainStoryInfo from '../content/mainStoryInfo';
```

**Review by BrookeGraham — CHANGES_REQUESTED** [review]
packages/story-editor/src/components/publishModal/test/storyPreview.js needs update import as well. line 27.
`import StoryPreview from '../content/storyPreview';`

**Comment by BrookeGraham** [comment]
I can make a ticket tomorrow but I wanted to flag this tooltip edge case. When the tooltip is on the publish modal and you scroll the anchor out of the modal, the tooltip stays mounted.

tooltip modal

**Comment by BrittanyIRL** [comment]
> When the tooltip is on the publish modal and you scroll the anchor out of the modal, the tooltip stays mounted.

Oh good catch! I made a ticket, since github is working for me and down for you :D 
https://github.com/GoogleForCreators/web-stories-wp/issues/10673

File: packages/story-editor/src/components/publishModal/test/content.js (hunk 1 above)
Comment (BrittanyIRL) [reply packages/story-editor/src/components/publishModal/test/content.js:38]: ah yup, i totally missed this file. thank you!

**Comment by BrittanyIRL** [comment]
> packages/story-editor/src/components/publishModal/test/storyPreview.js needs update import as well. line 27. `import StoryPreview from '../content/storyPreview';`

thanks for catching these, totally missed them. should be all fixed now.

**Review by BrookeGraham — APPROVED** [review]
😍

**Pull request merged**